evaporate.js

```diff
@@ -1273,12 +1273,14 @@
 
         function uploadKey(fileUpload) {
             // The key tries to give a signature to a file in the absence of its path.
-            // "<filename>-<mimetype>-<modifieddate>-<filesize>"
+            // Need to include path (if present) in case the same file is present in multiple paths/folders
+            // "<filename>-<mimetype>-<modifieddate>-<filesize>-<path>"
             return [
                 fileUpload.file.name,
                 fileUpload.file.type,
                 dateISOString(fileUpload.file.lastModifiedDate),
-                fileUpload.file.size
+                fileUpload.file.size,
+                fileUpload.name
```

Comment:
If you leave `allowS3ExistenceOptimization` to `false`, EvaporateJS will not attempt to perform this optimization and will simply re-upload the file, which is the desired behavior given how this PR was described.

FWIW, the RACR optimization related to incomplete and uploaded parts should still be available.